src/control/lib/control/pool.go

```diff
@@ -378,6 +378,8 @@ type (
 		Leader          uint32               `json:"leader"`
 		Rebuild         *PoolRebuildStatus   `json:"rebuild"`
 		TierStats       []*StorageUsageStats `json:"tier_stats"`
+		InfoBit         int32                `json:"info_bit"`
+		RankSet         string               `json:"rank_set"`
```

Comment:
I believe our internal type HostList can be used to produce a nicely formatted list of ranks similar to what the rank_set printing produces.

Reply:
Not sure to fully understand your remark: the idea should be to change the RankSet type to a list of ranks?
If yes, we are currently relying on the C code of the engine which produce an already formatted string of the ranks. With the time constraints that we have on this PR, it will be difficult to make this modification in this PR. As suggested in the previous comment, I could define a ticket to handle this development and add the reference of this new ticket in the code.

Reply:
My thought was that since RankSet is a field associated with the control API, it should take advantage of our Go specific types. You can use `hostlist.CreateNumericList` to create a HostList type from a string that represents a list of numbers (i.e. a rank list). The compact string format passed up from the engine is almost certainly compatible.

Reply:
- [ ] Convert C string to hostlist with `hostlist.CreateNumericList()`